Library/ViewModels/DiscoveryPostcardViewModel.swift

```diff
@@ -155,93 +182,184 @@ public final class DiscoveryPostcardViewModel: DiscoveryPostcardViewModelType,
 
-    self.projectImageURL = project.map { $0.photo.full }.map(URL.init(string:))
+    self.projectImageURL = configuredProject.map { $0.photo.full }.map(URL.init(string:))
 
-    self.projectNameAndBlurbLabelText = project
+    self.projectNameAndBlurbLabelText = configuredProject
       .map(projectAttributedNameAndBlurb)
 
-    self.projectStateIconHidden = project
+    self.projectStateIconHidden = configuredProject
       .map { $0.state != .successful }
 
-    self.projectStateStackViewHidden = project.map { $0.state == .live }.skipRepeats()
+    self.projectStateStackViewHidden = configuredProject.map { $0.state == .live }.skipRepeats()
 
-    self.projectStateSubtitleLabelText = project
+    self.projectStateSubtitleLabelText = configuredProject
       .map {
         $0.state != .live
           ? Format.date(secondsInUTC: $0.dates.stateChangedAt, dateStyle: .medium, timeStyle: .none)
           : ""
     }
 
-    self.projectStateTitleLabelColor = project
+    self.projectStateTitleLabelColor = configuredProject
       .map { $0.state == .successful ? .ksr_text_green_700 : .ksr_text_navy_700 }
       .skipRepeats()
 
-    self.projectStateTitleLabelText = project
+    self.projectStateTitleLabelText = configuredProject
       .map(fundingStatusText(forProject:))
 
     self.projectStatsStackViewHidden = self.projectStateStackViewHidden.map(negate)
 
-    self.socialImageURL = project
+    self.socialImageURL = configuredProject
       .map { $0.personalization.friends?.first.flatMap { URL(string: $0.avatar.medium) } }
 
-    self.socialLabelText = project
+    self.socialLabelText = configuredProject
       .map { $0.personalization.friends.flatMap(socialText(forFriends:)) ?? "" }
 
-    self.socialStackViewHidden = project
+    self.socialStackViewHidden = configuredProject
       .map { $0.personalization.friends == nil || $0.personalization.friends?.count ?? 0 == 0 }
       .skipRepeats()
 
-    self.fundingProgressContainerViewHidden = project
+    self.fundingProgressContainerViewHidden = configuredProject
       .map { $0.state == .canceled || $0.state == .suspended }
 
-    self.fundingProgressBarViewHidden = project
+    self.fundingProgressBarViewHidden = configuredProject
       .map { $0.state == .failed }
 
-    self.notifyDelegateShareButtonTapped = project
+    self.notifyDelegateShareButtonTapped = configuredProject
       .map(ShareContext.discovery)
       .takeWhen(self.shareButtonTappedProperty.signal)
 
+    let loggedOutUserTappedSaveButton = currentUser
+      .takeWhen(self.saveButtonTappedProperty.signal)
+      .filter(isNil)
+      .ignoreValues()
+
+    let loggedInUserTappedSaveButton = currentUser
+      .takeWhen(self.saveButtonTappedProperty.signal)
+      .filter(isNotNil)
+      .ignoreValues()
+
+    let userLoginAfterTappingSaveButton = Signal.combineLatest(
+      self.userSessionStartedProperty.signal,
+      loggedOutUserTappedSaveButton
+      )
+      .ignoreValues()
+      .take(first: 1)
```

Comment:
Curious why there is a `.take(first: 1)` here? I believe if the user triggered this signal, then logged out, then logged in, and then tried to save a project, this would never emit again. Weird use case, I know, but still...